engine/src/desktop.cpp

```diff
@@ -353,7 +353,11 @@ void MCPlatformHandleMouseMove(MCPlatformWindowRef p_window, MCPoint p_location)
 
 void MCPlatformHandleMouseDown(MCPlatformWindowRef p_window, uint32_t p_button, uint32_t p_count)
 {
-	if (((MCScreenDC *)MCscreen) -> isbackdrop(p_window))
+    // ensure the dragboard is bound correctly
+    MCAutoRefcounted<MCRawClipboard> t_dragboard(MCRawClipboard::CreateSystemDragboard());
```

Comment:
I wonder if this should be further down... Around line 394 - I think that's where the 'reset for potential drag' happens...

Reply:
Indeed - perhaps instead of MCdragboard->Clear()?

Reply:
hehe.. tbh I didn't look that much at the rest of `MCPlatformHandleMouseDown` once I moved it there and things worked. Yes it should replace that of course!